mypy/join.py

```diff
@@ -391,7 +391,17 @@ def combine_similar_callables(t: CallableType, s: CallableType) -> CallableType:
 
 
 def combine_arg_names(t: CallableType, s: CallableType) -> List[Optional[str]]:
-    # Note: assumes is_similar_type(t, s) is true
+    """Takes two callables and produces the meet of their argument names.
```

Comment:
Extreme nit: this calls this operation a meet in one place and a join in another, and should use the same term twice. I think it is more of a meet than a join (since it is like an intersection, which are meets?). Just saying "combination" or something fine also.

Reply:
Oh whoops, that's embarrassing -- I didn't mean to use the word 'meet' here...

But yeah, I agree it's confusing, especially since this function isn't even manipulating types. I reworded the comment completely and just used the phrase "compatible with" which is hopefully better.